hazelcast/src/main/java/com/hazelcast/cardinality/operations/AggregateAndEstimateOperation.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cardinality.operations;
+
+import com.hazelcast.cardinality.CardinalityEstimatorContainer;
+import com.hazelcast.cardinality.CardinalityEstimatorDataSerializerHook;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.spi.Operation;
+
+import java.io.IOException;
+
+public class AggregateAndEstimateOperation
+        extends CardinalityEstimatorBackupAwareOperation {
+
+    private long hash;
+    private long estimate;
+
+    public AggregateAndEstimateOperation() { }
+
+    public AggregateAndEstimateOperation(String name, long hash) {
+        super(name);
+        this.hash = hash;
+    }
+
+    @Override
+    public int getId() {
+        return CardinalityEstimatorDataSerializerHook.AGGREGATE_AND_ESTIMATE;
+    }
+
+    @Override
+    public void run() throws Exception {
+        CardinalityEstimatorContainer est = getCardinalityEstimatorContainer();
```

Comment:
I would name this `container` like I saw it in other operations, to prevent confusion with the field `estimate`.